pkg/profilecreator/profilecreator.go

```diff
@@ -229,22 +229,43 @@ func (ghwHandler GHWHandler) CPU() (*cpu.Info, error) {
 	return ghw.CPU(ghwHandler.snapShotOptions)
 }
 
-// Topology returns a TopologyInfo struct that contains information about the Topology on the host system
-func (ghwHandler GHWHandler) Topology() (*topology.Info, error) {
-	return ghw.Topology(ghwHandler.snapShotOptions)
+// SortedTopology returns a TopologyInfo struct that contains information about the Topology on the host system
+func (ghwHandler GHWHandler) SortedTopology() (*topology.Info, error) {
+	topologyInfo, err := ghw.Topology(ghwHandler.snapShotOptions)
+	if err != nil {
+		return nil, fmt.Errorf("Error obtaining Topology Info from GHW snapshot: %v", err)
+	}
+	// Sorting the NUMA nodes
+	sort.Slice(topologyInfo.Nodes, func(x, y int) bool {
+		return topologyInfo.Nodes[x].ID < topologyInfo.Nodes[y].ID
+	})
+	for _, node := range topologyInfo.Nodes {
+		//Sorting the logical cores in a processor
+		for _, core := range node.Cores {
+			sort.Slice(core.LogicalProcessors, func(x, y int) bool {
+				return core.LogicalProcessors[x] < core.LogicalProcessors[y]
+			})
+		}
+		//Sorting by comparing the first logical core of processor cores
+		sort.Slice(node.Cores, func(i, j int) bool {
+			return node.Cores[i].LogicalProcessors[0] < node.Cores[j].LogicalProcessors[0]
+		})
+	}
+	return topologyInfo, nil
```

Comment:
this function deserves a (couple of) unit test - can be added later.

Reply:
Sure, will add the tests shortly.